### PR TITLE
Extractor + Headless Actions structures to enums

### PR DESCRIPTION
**Pull request by LuitelSamikshya**
## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by Ice3man543 — COMMENTED** [review]
As an extractor can only have a single defined value, I think it makes sense for us to convert `ExtractorTypes` from `[]ExtractorType` to a single `ExtractorType` value which can only hold one type. Also, do you think it'd make sense to define this extractor type related functionality in a separate file?

That way, once we've added this, we can similar add enums to other parts of the codebase as well.

File: v2/pkg/operators/extractors/extractors.go [hunk 1]
```diff
@@ -126,17 +128,45 @@ const (
...
+}
+
+func toExtractorTypes(valueToMap string) (ExtractorType, error) {
+	normalizedValue := normalizeValue(valueToMap)
+	for key, currentValue := range extractorMappings {
+		if normalizedValue == currentValue {
+			return key, nil
+		}
+	}
+	return -1, errors.New("Invalid exctractor type: " + valueToMap)
```
Comment (Ice3man543) [inline v2/pkg/operators/extractors/extractors.go:163]: typo

Comment (LuitelSamikshya) [reply v2/pkg/operators/extractors/extractors.go]: Done

File: v2/pkg/operators/extractors/compile.go [hunk 2]
```diff
@@ -48,7 +47,7 @@ func (e *Extractor) CompileExtractors() error {
 	}
 
 	if e.CaseInsensitive {
-		if e.Type != "kval" {
+		if e.GetType().String() != "kval" {
```
Comment (forgedhallpass) [inline v2/pkg/operators/extractors/compile.go:50]: It should rather be `e.GetType() != KValExtractor`. The point is not to rely on hard-coded string comparisons

**Comment by forgedhallpass** [comment]
It seems that this PR only handles one struct, whereas the related issue referred to the whole code base. Please identify all the missing places and add it in the ticket as tasks, then explicitly mention which task this PR resolves.

File: v2/pkg/operators/extractors/compile.go (hunk 2 above)
Comment (LuitelSamikshya) [reply v2/pkg/operators/extractors/compile.go]: Done

File: v2/pkg/operators/extractors/compile.go [hunk 3]
```diff
@@ -10,13 +10,13 @@ import (
 
 // CompileExtractors performs the initial setup operation on an extractor
 func (e *Extractor) CompileExtractors() error {
-	var ok bool
+	//var ok bool
```
Comment (Mzack9999) [inline v2/pkg/operators/extractors/compile.go:13]: Could you please remove the commented code?

**Review by Mzack9999 — CHANGES_REQUESTED** [review]

File: v2/pkg/operators/extractors/compile.go (hunk 3 above)
Comment (LuitelSamikshya) [reply v2/pkg/operators/extractors/compile.go]: Done

**Review by Mzack9999 — APPROVED** [review]

**Pull request merged**